src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/ProjectPropertiesProviders/DelegatedProjectPropertiesBase.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.ProjectPropertiesProviders
+{
+    /// <summary>
+    /// An implementation of IProjectProperties that delegates its operations
+    /// to another IProjectProperties object
+    /// </summary>
+    internal class DelegatedProjectPropertiesBase : IProjectProperties
+    {
+        protected internal IProjectProperties DelegatedProperties;
+
+        public DelegatedProjectPropertiesBase(IProjectProperties properties)
+        {
+            DelegatedProperties = properties;
+        }
+
+        /// <summary>
+        /// <see cref="IProjectProperties.Context"/>
+        /// </summary>
+        public virtual IProjectPropertiesContext Context => DelegatedProperties.Context;
+
+        /// <summary>
+        /// <see cref="IProjectProperties.FileFullPath"/>
+        /// </summary>
+        public virtual string FileFullPath => DelegatedProperties.FileFullPath;
+
+        /// <summary>
+        /// <see cref="IProjectProperties.PropertyKind"/>
```

Comment:
@davkean maybe that's a good feature for C# compiler to make `<summary>` optional in XML comments.

F# compiler support this, it would look like this:

```fsharp
/// <see cref="IProjectProperties.Context"/>
override x.Context = DelegatedProperties.Context
```